var/spack/repos/builtin/packages/intel-oneapi-compilers/package.py

```diff
@@ -83,6 +83,10 @@ class IntelOneapiCompilers(IntelOneApiPackage):
                  placement='fortran-installer',
                  when='@2021.1.2')
 
+    variant('CC', default='icx',
+            values=('icx', 'icc', 'dpcpp'), multi=False,
+            description='Set CC & friends in run environment')
```

Comment:
There's a conflict of outcomes here. On the one hand, we want to create module files that set CC etc. for convenience; to do that this PR adds a CC variant. While that is helpful, it also means that each module people want to generate will result in a separate installations of the `intel-oneapi-compilers` package.

I think it might be preferable to create a package which depends on `intel-oneapi-compilers` called something like `intel-oneapi-icc-wrappers`, and *that* package can create a module which sets `CC=icc` etc. - does that seem reasonable?

Side note: This seems to relate to `use_variant`, but IMO, a user who explicitly wants to access these different compiler executables may in fact benefit from an explicit `-wrapper`-style package.

Reply:
@mathomp4: What do you think about @scheibelp suggestion. I don't use module files and do not know if it is important that the package name be intel-oneapi-compilers

Reply:
@rscohn2 Hmm. I don't know if it would matter too much. For my application that I'm hoping to bring into spack "soon" (I have to submit about 6 new package PRs first), just a few `module load` commands bring in like 30-40 modules as LMod does its dependency magic.

So probably in the end, I'd present to users a sort of "meta-module" that loads everything underneath. I could call *that* module anything in regards to the compiler name as long as I know what the "real" intel module is called.